protocol/communities/manager.go

```diff
@@ -1812,22 +1958,42 @@ func (m *Manager) HandleCommunityRequestToJoin(signer *ecdsa.PublicKey, request
 	// More specifically, CommunityRequestToLeave may be delivered later than CommunityRequestToJoin, or not delivered at all
 	acceptAutomatically := community.AcceptRequestToJoinAutomatically() || community.HasMember(signer)
 	if acceptAutomatically {
-		err = m.markRequestToJoin(signer, community)
-		if err != nil {
-			return nil, err
+		if community.IsControlNode() {
+			err = m.markRequestToJoin(signer, community)
+			if err != nil {
+				return nil, err
+			}
+			// Don't check permissions here,
+			// it will be done further in the processing pipeline.
+			requestToJoin.State = RequestToJoinStateAccepted
+		} else {
+			err = m.markRequestToJoinAsAcceptedPending(signer, community)
+			if err != nil {
+				return nil, err
+			}
+			requestToJoin.State = RequestToJoinStateAcceptedPending
 		}
-		// Don't check permissions here,
-		// it will be done further in the processing pipeline.
-		requestToJoin.State = RequestToJoinStateAccepted
 		return requestToJoin, nil
 	}
 
-	permissionsSatisfied, _, err := m.accountsSatisfyPermissionsToJoin(community, request.RevealedAccounts)
-	if err != nil {
-		return nil, err
-	}
-	if !permissionsSatisfied {
-		requestToJoin.State = RequestToJoinStateDeclined
+	if community.IsControlNode() && len(request.RevealedAccounts) > 0 {
```

Comment:
if `len(request.RevealedAccounts) == 0` and the community has permissions to join, we should auto set it to `RequestToJoinStateDeclined`. I'm not sure if the code does this.

Reply:
Ah actually, the check on `len(request.RevealedAccounts)` is a left over but it probably wouldn't hurt. It ends up being marked as `Declined` a few line below latest after the satisfaction check was done